### ISSUE TITLE
FolderBase Dataset automatically resolves under current directory when data_dir is not specified

### ISSUE THREAD

**Issue by npuichigo** [huggingface/datasets#6152]
### Describe the bug

FolderBase Dataset automatically resolves under current directory when data_dir is not specified.

For example:
```
load_dataset("audiofolder")
```
takes long time to resolve and collect data_files from current directory. But I think it should reach out to this line for error handling https://github.com/huggingface/datasets/blob/cb8c5de5145c7e7eee65391cb7f4d92f0d565d62/src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py#L58-L59

### Steps to reproduce the bug

```
load_dataset("audiofolder")
```

### Expected behavior

Error report

### Environment info

- `datasets` version: 2.14.4
- Platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.17
- Python version: 3.8.15
- Huggingface_hub version: 0.16.4
- PyArrow version: 12.0.1
- Pandas version: 1.5.3

**Comment by npuichigo** [comment]
@lhoestq

**Comment by lhoestq** [comment]
Makes sense, I guess this can be fixed in the load_dataset_builder method.
It concerns every packaged builder I think (see values in `_PACKAGED_DATASETS_MODULES`)

**Comment by npuichigo** [comment]
I think the behavior is related to these lines, which short circuited the error handling.
https://github.com/huggingface/datasets/blob/664a1cb72ea1e6ef7c47e671e2686ca4a35e8d63/src/datasets/load.py#L946-L952

So should data_dir be checked here or still delegating to actual `DatasetModule`? In that case, how to properly set `data_files` here.

**Comment by lhoestq** [comment]
This is location in PackagedDatasetModuleFactory.get_module seems the be the right place to check if at least data_dir or data_files are passed

**Comment by debrupf2946** [comment]
@mariosasko can you please assign this issue to me,I want to work on this

**Comment by debrupf2946** [comment]
#self-assign

**Comment by zutarich** [comment]
@mariosasko is this issue still open? i would love to kickstart my journey to open source with this issue!
Regards
zutarich

**Comment by mariosasko** [comment]
It is unless @debrupf2946 is working on it.

**Comment by Etelis** [comment]
#self-assign

**Comment by JINO-ROHIT** [comment]
@self-assign

**Comment by mariosasko** [comment]
Yes, feel free to work on this :)

**Comment by JINO-ROHIT** [comment]
i think its working as expected . Heres the log i get for the same line -

![image](https://github.com/huggingface/datasets/assets/63234112/8a857ec5-8dd0-4b01-b3a7-7c93444b9558)

**Comment by sahillihas** [comment]
#self-assign

**Comment by aniyagnik** [comment]
Hey! I saw that PR #7314 fixed the issue when no matching data files are found, which is great. But I think the builder still defaults to the current directory when `data_dir` isn't specified, which could lead to unexpected behavior.

If no one’s actively working on that part, I’d be happy to pick it up and propose a fix. Let me know if that works!

**Comment by lhoestq** [comment]
Cool, feel free to ping me if you open a PR @aniyagnik :)

**Comment by ArjunJagdale** [comment]
Hey! I noticed this is still open and unresolved. I’d like to take this up and submit a PR to prevent resolving to the current directory when data_dir is not specified!!!

**Comment by ArjunJagdale** [comment]
Hi @aniyagnik @npuichigo @sahillihas @lhoestq @mariosasko,
I've opened a PR to address the issue discussed in #6152 by adding an early validation check for folder-based datasets. Would appreciate your review whenever convenient. Thanks!

https://github.com/huggingface/datasets/pull/7618
